deploy/runtools/run_farm.py

```diff
@@ -804,3 +810,500 @@ def terminate_by_inst(self, inst: Inst) -> None:
             f"WARNING: Skipping terminate_by_inst since run hosts are externally provisioned."
         )
         return
+
+class LocalProvisionedVM(RunFarm): # run_farm_type
+    """Manages the set of locally provisioned VMs with FPGAs passed through to them. This class is responsible for spinning up the VMs and shutting down those VMs after the simulation is done.
+
+    Attributes:
+        vm_ip_to_instance: dict of IP address to instance name
+    """
+
+    run_farm_hosts_dict: Dict[
+        str, List[Tuple[Inst, Optional[Union[EC2InstanceResource, MockBoto3Instance, str]]]]
+    ]
+    vm_name: str = f"jammy_cis-{local('whoami', capture=True)}"
+    vm_username: str = local("whoami", capture=True)
```

Comment:
I'm not sure I would call the `whoami` here? Maybe just run it once in the `__init__`.